pandas/core/groupby/groupby.py

```diff
@@ -1924,10 +1992,15 @@ def _get_cythonized_result(self, how, grouper, aggregate=False,
             Whether the result of the Cython operation is an index of
             values to be retrieved, instead of the actual values themselves
         pre_processing : function, default None
-            Function to be applied to `values` prior to passing to Cython
-            Raises if `needs_values` is False
+            Function to be applied to `values` prior to passing to Cython.
+            Function should return a tuple where the first element is the
+            values to be passed to Cython and the second element is an
+            optional dictionary containing type inferences to be applied
+            after the Cython call. Raises if `needs_values` is False
         post_processing : function, default None
-            Function to be applied to result of Cython function
+            Function to be applied to result of Cython function. Should accept
```

Comment:
maybe you want to show the signature as a comment of the Callable[object, dict]

Reply:
Are you asking to do that in this docstring or as a type hint in the implementations themselves?

Reply:
both